### PR TITLE
Allow negation of ignore and only patterns.

### PR DESCRIPTION
**Pull request by loganfsmyth**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | N
| Major: Breaking Change?  | N
| Minor: New Feature?      | Y
| Deprecations?            | 
| Spec Compliancy?         | 
| Tests Added/Pass?        | 
| Fixed Tickets            |  
| License                  | MIT
| Doc PR                   | 
| Dependency Changes       | 

Only handles negations within a single pattern set so if you add some negation lines, it won't negate patterns from other `ignore` or `only` blocks.

### CODE REVIEW

File: packages/babel-core/src/config/build-config-chain.js [hunk 1]
```diff
@@ -46,63 +46,63 @@ class ConfigChainBuilder {
...
-    if (ignore) {
-      if (!Array.isArray(ignore)) {
-        throw new Error(`.ignore should be an array, was ${JSON.stringify(ignore)}`);
-      }
-
-      for (const pattern of ignore) {
-        if (this.matchesPattern(pattern, dirname)) return true;
-      }
+    if (ignore && !Array.isArray(ignore)) {
+      throw new Error(`.ignore should be an array, was ${JSON.stringify(ignore)}`);
```
Comment (xtuc) [inline packages/babel-core/src/config/build-config-chain.js]: I don't like the `was ..`. What about:

```
.ignore should be an array, "foobar" given
```

File: packages/babel-core/test/config-chain.js [hunk 2]
```diff
@@ -557,6 +557,55 @@ describe("buildConfigChain", function () {
...
+        alias: fixture(".babelignore"),
+        loc: fixture(".babelignore"),
+        dirname: fixture(),
+      },
+      {
+        type: "options",
+        options: {
+          ignore: [
+            "*",
+            "!src.js",
```
Comment (xtuc) [inline packages/babel-core/test/config-chain.js]: Could we add an test with a folder?

Comment (loganfsmyth) [inline packages/babel-core/test/config-chain.js]: Done.

File: packages/babel-core/src/config/build-config-chain.js (hunk 1 above)
Comment (loganfsmyth) [inline packages/babel-core/src/config/build-config-chain.js]: Done.

**Pull request merged**